livestream-compose-sample/src/main/java/io/getstream/livestream/compose/ui/LiveStreamActivity.kt

```diff
@@ -0,0 +1,137 @@
+package io.getstream.livestream.compose.ui
+
+import android.Manifest
+import android.content.ClipboardManager
+import android.content.Context
+import android.content.Intent
+import android.content.pm.PackageManager
+import android.os.Bundle
+import android.widget.Toast
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.activity.viewModels
+import androidx.core.app.ActivityCompat
+import androidx.core.content.ContextCompat
+import io.getstream.chat.android.client.ChatClient
+import io.getstream.chat.android.compose.ui.theme.ChatTheme
+import io.getstream.chat.android.compose.viewmodel.messages.MessageComposerViewModel
+import io.getstream.chat.android.compose.viewmodel.messages.MessageListViewModel
+import io.getstream.chat.android.compose.viewmodel.messages.MessagesViewModelFactory
+import io.getstream.chat.android.offline.ChatDomain
+import io.getstream.livestream.compose.R
+import io.getstream.livestream.compose.models.LiveStreamType
+import io.getstream.livestream.compose.streams.CameraLiveStream
+import io.getstream.livestream.compose.streams.VideoLiveStream
+import io.getstream.livestream.compose.streams.YoutubeLiveStream
+
+class LiveStreamActivity : ComponentActivity() {
+
+    private val factory by lazy {
+        MessagesViewModelFactory(
+            this,
+            getSystemService(CLIPBOARD_SERVICE) as ClipboardManager,
+            ChatClient.instance(),
+            ChatDomain.instance(),
+            intent.getStringExtra(KEY_CHANNEL_ID) ?: "",
+            30
+        )
+    }
+
+    private val composerViewModel by viewModels<MessageComposerViewModel>(factoryProducer = { factory })
+    private val listViewModel by viewModels<MessageListViewModel>(factoryProducer = { factory })
+    private var liveStreamType: LiveStreamType = LiveStreamType.Youtube
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        handleExtras()
+        setViewContent()
+    }
+
+    override fun onRequestPermissionsResult(
+        requestCode: Int, permissions: Array<String>, grantResults: IntArray
+    ) {
+        super.onRequestPermissionsResult(requestCode, permissions, grantResults)
+        if (requestCode == REQUEST_CODE_PERMISSIONS) {
+            if (allPermissionsGranted()) {
+                setViewContent()
+            } else {
+                Toast.makeText(
+                    this,
+                    getString(R.string.permission_message),
+                    Toast.LENGTH_SHORT
+                ).show()
+                finish()
+            }
+        }
+    }
+
+    private fun setViewContent() {
```

Comment:
I think we shouldn't do this - we should call `setContent()` in `onCreate()` and then pass in a custom function that represents this screen, e.g. `LiveStreamContent()`. Just like with all of our other examples.

Reply:
WIll request another pass on new changes now esp after camera permissions bit